crates/ntx-builder/src/state/mod.rs

```diff
@@ -70,10 +87,24 @@ pub struct State {
 }
 
 impl State {
+    /// The number of blocks to keep in memory while tracking the chain tip.
+    const CHAIN_TIP_LENGTH: u32 = 128;
```

Comment:
@Mirko-von-Leipzig unsure what length we want and whether we want it configurable.

Reply:
I think this depends on whether we can avoid cloning the struct for every transaction. If we can avoid cloning, we can make it fairly long (e.g., 128 blocks or maybe even more), if not, I'd stick on the smaller side - e.g., 16 blocks.